modules/artifact/src/main/java/org/mule/module/artifact/classloader/GroovyResourceReleaser.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.artifact.classloader;
+
+import static org.mule.runtime.core.api.util.ClassUtils.getFieldValue;
+import static org.mule.runtime.core.api.util.ClassUtils.getStaticFieldValue;
+import static org.mule.runtime.core.api.util.ClassUtils.loadClass;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.mule.runtime.module.artifact.api.classloader.ResourceReleaser;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+
+import org.slf4j.Logger;
+
+/**
+ * A utility class to release all resources associated to Groovy Dependency on un-deployment to prevent classloader leaks.
+ *
+ * @since 4.4.0
```

Comment:
```suggestion
 * @since 4.3.0
```
We'll need to change this if we backport it to 4.3.